### PR TITLE
[BugFix] Handle parquet exception caused by wrong s3 configuration

### PR DESCRIPTION
**Pull request by letian-jiang**
## Problem Summary:
Fixes # (issue)

Using a wrong s3 configuration (ak/sk/endpoint) will not be checked until the parquet writer tries to flush buffered data into the output stream. In this case, `BufferedPageWriter.Close()` throws an exception, which should be handled.

Moreover, the deconstructor of `ParquetFileWriter` causes SIGSEGV. We have to mark the file writer as closed and release its ownership to avoid calling its dtor.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/formats/parquet/file_writer.cpp [hunk 1]
```diff
@@ -371,24 +371,41 @@ Status FileWriterBase::split_offsets(std::vector<int64_t>& splitOffsets) const {
...
-        _chunk_writer = nullptr;
+        try {
+            _chunk_writer->close();
+        } catch (const ::parquet::ParquetStatusException& e) {
+            // this is to avoid ParquetFileWriter.Close which cause segfaults
+            _closed = true;
+            _writer.release();
+
+            auto st = Status::IOError(e.what());
+            st = st.clone_and_prepend("flush row group error");
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:384]:
```c++
std::string msg = e.what() + " flush row group error"
auto st = Status::IOError(msg);
LOG(WARNING) << msg;
return st;
```
better?

File: be/src/formats/parquet/file_writer.cpp [hunk 2]
```diff
@@ -371,24 +371,41 @@ Status FileWriterBase::split_offsets(std::vector<int64_t>& splitOffsets) const {
...
+    return Status::OK();
 }
 
 Status SyncFileWriter::close() {
     if (_closed) {
         return Status::OK();
     }
 
-    _flush_row_group();
+    auto st = _flush_row_group();
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:399]: just `ASSIGN_OR_RETURN(auto st, _flush_row_group())`.

File: be/src/formats/parquet/file_writer.cpp [hunk 3]
```diff
@@ -371,24 +371,41 @@ Status FileWriterBase::split_offsets(std::vector<int64_t>& splitOffsets) const {
...
+    RETURN_IF_ERROR(_flush_row_group());
     _writer->Close();
 
-    auto st = _outstream->Close();
-    if (!st.ok()) {
-        return Status::InternalError("Close file failed!");
+    auto arrow_st = _outstream->Close();
+    if (!arrow_st.ok()) {
+        st = Status::IOError(arrow_st.message());
+        st = st.clone_and_prepend("close file error");
```
Comment (Smith-Cruise) [inline be/src/formats/parquet/file_writer.cpp:406]: ditto

File: be/src/formats/parquet/file_writer.cpp (hunk 1 above)
Comment (letian-jiang) [reply be/src/formats/parquet/file_writer.cpp]: done

File: be/src/formats/parquet/file_writer.cpp (hunk 2 above)
Comment (letian-jiang) [reply be/src/formats/parquet/file_writer.cpp]: updated

File: be/src/formats/parquet/file_writer.cpp (hunk 3 above)
Comment (letian-jiang) [reply be/src/formats/parquet/file_writer.cpp]: done

**Review by Smith-Cruise — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by stephen-shelby — DISMISSED** [review]

File: be/src/formats/parquet/file_writer.cpp [hunk 4]
```diff
@@ -371,24 +371,38 @@ Status FileWriterBase::split_offsets(std::vector<int64_t>& splitOffsets) const {
...
-        _chunk_writer = nullptr;
+        try {
+            _chunk_writer->close();
+        } catch (const ::parquet::ParquetStatusException& e) {
+            // this is to avoid ParquetFileWriter.Close which cause segfaults
+            _closed = true;
+            _writer.release();
+
+            auto st = Status::IOError(fmt::format("{}: {}", "flush rowgroup error", e.what()));
+            LOG(WARNING) << st;
```
Comment (Youngwb) [inline be/src/formats/parquet/file_writer.cpp:384]: add `_chunk_writer.reset(); ` before return?

Comment (letian-jiang) [reply be/src/formats/parquet/file_writer.cpp:386]: done

**Review by Youngwb — APPROVED** [review]

**Review by Smith-Cruise — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**